includes/Generator/Customer.php

```diff
@@ -15,23 +15,23 @@ class Customer extends Generator {
 	/**
 	 * Return a new customer.
 	 *
-	 * @param bool $save Save the object before returning or not.
+	 * @param bool   $save Save the object before returning or not.
+	 * @param string $emailDomain An optional domain to be used for all customers' email addresses.
 	 * @return \WC_Customer Customer object with data populated.
 	 */
-	public static function generate( $save = true ) {
+	public static function generate( $save = true, $emailDomain = '' ) {
 		self::init_faker();
 
-		// Make sure a unique username and e-mail are used.
-		do {
-			$username = self::$faker->userName();
-		} while ( username_exists( $username ) );
+		$safeEmailDomain = $emailDomain ? $emailDomain : self::$faker->safeEmailDomain();
```

Comment:
The second instance of the variable is redundant. 

```suggestion
		$safeEmailDomain = $emailDomain ?: self::$faker->safeEmailDomain();
```